src/Streamly/Internal/LZ4.hs

```diff
@@ -484,6 +484,15 @@ decompressResizedD (Stream.Stream step0 state0) =
         r <- step0 gst st
         case r of
             Stream.Yield arr st1 -> do
+              compLen32 <-
+                  liftIO $ Array.asPtr arr $ \src ->
+                      peek (castPtr src `plusPtr` 4 :: Ptr Int32)
+              if compLen32 <= 0
+              then do
+                let (Array.Array fptr end) = arr
+                    arr1 = Array.Array (fptr `plusForeignPtr` 8) end
+                return $ Stream.Yield arr1 (DecompressDo st1 lz4Ctx prev)
+              else do
```

Comment:
I think we should do this inside `decompressChunk` itself instead of doing it outside.